### PR TITLE
Add BlockCounter QML component which handles NotifyBlockTip signal

### PR DESCRIPTION
**Pull request by hebasto**
This PR:
1) introduces additional `components` sub-directory for QML objects for basic GUI elements
2) adds a prototype of QML object which displays block tip height
3) handles the `NotifyBlockTip` signal from the core

---

From Qt [docs](https://doc.qt.io/qt-5.15/qmlapplications.html):
> QML is a declarative language that allows user interfaces to be described in terms of their visual components and how they interact and relate with one another. It is a highly readable language that was designed to enable components to be interconnected in a dynamic manner, and it allows components to be easily reused and customized within a user interface.
> ...
> It allows **developers _and_ designers** alike to create highly performant, fluidly animated and visually appealing applications.

#### Review notes for developers

This PR continues to exploit the technique of embedding an instance of the `NodeModel` class into QML code as a context property (see #11). This approach differs from alternative ones, in a way that there are no signal handlers for signals from C++ code in the QML files -- only properties bindings are used.

The benefits of such an approach are:
- QML files are easy to read, reason about, and modify. Designers could easy work on them
- JavaScript code assignments in signal handlers break properties bindings

Exposing properties of the `NodeModel` class to QML code justifies decision to not re-use the `ClientModel` class because these classes have different designs.

#### Review notes for designers

This PR demonstrates an idea of small QML files that describe GUI elements and their interaction in a _declarative_ way.

You could easy and safely tinker with object properties. Just run `make clean && make` again to see the result of your changes.

Names of QML files in the `components` sub-directory are actually the names of defined QML objects. Therefore, to make your life easier, you could suggest names you are comfortable with.

Don't hesitate to request new QML files that describe your design ideas :)

### CODE REVIEW

**Comment by hebasto** [comment]
@jonasschnelli @Talkless Do you mind reviewing this PR at a concept level?

**Comment by hebasto** [comment]
Updated 7cc9c80ac80b245d75e16e87e5926716ee6a7ac0 -> 5467df4cd0e29794d38d1951fc9cb87e718a7809 ([pr31.01](https://github.com/hebasto/gui-qml/commits/pr31.01) -> [pr31.02](https://github.com/hebasto/gui-qml/commits/pr31.02)):

- rebased due to the conflict with #25
- the `blockTipHeight` is read-only now

**Comment by hebasto** [comment]
Rebased 5467df4cd0e29794d38d1951fc9cb87e718a7809 -> e7b9a09fbdd9fa18ebd8bfd981421ecffde044e0 ([pr31.02](https://github.com/hebasto/gui-qml/commits/pr31.02) -> [pr31.03](https://github.com/hebasto/gui-qml/commits/pr31.03)) on top of the #28.

File: src/qml/components/BlockCounter.qml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+
+Rectangle {
+    color: "black"
+    width: label.width * 1.5
+
+    property int blockHeight
+
+    Label {
```
Comment (promag) [inline src/qml/components/BlockCounter.qml:10]: These controls have a `background` property:

```qml
Label {
    property int blockHeight: 0
    background: Rectangle {
        color: "black"
    }
    color: "orange"
    padding: 16
    text: blockHeight
}
```

File: src/qt/bitcoin.cpp [hunk 2]
```diff
@@ -85,7 +85,6 @@ static void RegisterMetaTypes()
 
     qRegisterMetaType<std::function<void()>>("std::function<void()>");
     qRegisterMetaType<QMessageBox::Icon>("QMessageBox::Icon");
-    qRegisterMetaType<interfaces::BlockAndHeaderTipInfo>("interfaces::BlockAndHeaderTipInfo");
```
Comment (promag) [inline src/qt/bitcoin.cpp:88]: Why is this removed?

**Comment by hebasto** [comment]
Updated e7b9a09fbdd9fa18ebd8bfd981421ecffde044e0 -> d473a47fe3e46bc68f946bb222b4e1b280016190 ([pr31.03](https://github.com/hebasto/gui-qml/commits/pr31.03) -> [pr31.04](https://github.com/hebasto/gui-qml/commits/pr31.04)):

- rebased due to the conflict with #34
- addressed @promag's [comment](https://github.com/bitcoin-core/gui-qml/pull/31#discussion_r698833036)

File: src/qml/components/BlockCounter.qml (hunk 1 above)
Comment (hebasto) [reply src/qml/components/BlockCounter.qml]: Thanks! [Updated](https://github.com/bitcoin-core/gui-qml/pull/31#issuecomment-908944132).

**Review by jarolrod — COMMENTED** [review]
ACK d473a47fe3e46bc68f946bb222b4e1b280016190

This feature works nicely and code is clean. The main thing to talk about here is the approach taken in terms of handling the Core node instance. 

Here, we have a certain layer of separation between the reference to the core node and what we do at the QML layer. This should be the approach we take going forward. It is scary to be passing around this reference around raw.

Additionally, here we are showcasing why using QML is a good approach for our project. There is great separation between our business logic and the UI logic. We can have our fellow designers live in one part of the codebase where programming is not required, while us developers focus on making things functional. To me, this showcases that we made the right decision.✨

**Comment by hebasto** [comment]
Trivially rebased d473a47fe3e46bc68f946bb222b4e1b280016190 -> 3ead757f60434933a915b0c726e58b419ea8b8c4 ([pr31.04](https://github.com/hebasto/gui-qml/commits/pr31.04) -> [pr31.05](https://github.com/hebasto/gui-qml/commits/pr31.05)) on top of the recent [sync](https://github.com/bitcoin-core/gui-qml/pull/36) with the main repo.

**Review by promag — COMMENTED** [review]
Code review ACK 3ead757f60434933a915b0c726e58b419ea8b8c4.

**Review by jarolrod — COMMENTED** [review]
ACK 3ead757 

Changes since my last review is only rebase over the weekly upstream sync

**Pull request merged**